src/simulation/LoadGenerator.cpp

```diff
@@ -1743,4 +1813,32 @@ GeneratedLoadConfig::modeUploads() const
     return mode == LoadGenMode::SOROBAN_UPLOAD ||
            mode == LoadGenMode::MIXED_CLASSIC_SOROBAN;
 }
+
+std::pair<TxGenerator::TestAccountPtr, TransactionFrameBaseConstPtr>
+LoadGenerator::readTransactionFromFile(GeneratedLoadConfig const& cfg)
+{
+    ZoneScoped;
+
+    // Read the next transaction from the file
+    TransactionEnvelope txEnv;
+    if (!mPreloadedTransactionsFile.readOne(txEnv))
+    {
+        // End of file reached, start again from the beginning
+        throw std::runtime_error("LoadGenerator: End of file reached");
```

Comment:
Hmm I don't think the comment is correct. Currently, if we throw it looks like loadgen will just fail because we don't catch this exception anywhere. Also, would this actually work? I think we'd have invalid sequence numbers if we started over.

Reply:
Good catch, the comment is wrong (leftover from the first iteration of this change). Yeah, we should not be re-using the same file. The idea is that we generate enough transactions at the beginning of the mission. Removed the comment.